external_webpage/data_source_reader.py

```diff
@@ -101,7 +104,7 @@ def read_config(self):
         """
 
         # read config
-        page = requests.get('http://{}:{}/'.format(self._host, PORT_CONFIG))
+        page = requests.get('http://{}:{}/'.format(self._host, PORT_CONFIG), timeout=URL_GET_TIMEOUT)
```

Comment:
If this times out an error is raised. This error should be handled. Furthermore, in the caller of this method the other error raise by this method should be caught and handled.

Reply:
See bottom of this page https://requests.readthedocs.io/en/master/user/quickstart/#make-a-request